srt-tokio/src/listener/request.rs

```diff
@@ -0,0 +1,95 @@
+use std::{io::ErrorKind, net::SocketAddr, time::Instant};
+
+use bytes::Bytes;
+use futures::{
+    channel::{mpsc, oneshot},
+    SinkExt,
+};
+use srt_protocol::settings::KeySettings;
+use srt_protocol::{connection::ConnectionSettings, listener::*, packet::*};
+
+use crate::{watch, SocketStatistics, SrtSocket};
+
+pub fn new_connection_request(
+    session_id: SessionId,
+    response_sender: mpsc::Sender<(SessionId, AccessControlResponse)>,
+    request: AccessControlRequest,
+    settings: oneshot::Receiver<ConnectionSettings>,
+    input_data: mpsc::Sender<(Instant, Bytes)>,
+    output_data: mpsc::Receiver<(Instant, Bytes)>,
+    statistics: watch::Receiver<SocketStatistics>,
+) -> ConnectionRequest {
+    ConnectionRequest {
+        response_sender: ResponseSender(session_id, response_sender),
+        request,
+        settings,
+        output_data,
+        input_data,
+        statistics,
+    }
+}
+
+#[derive(Debug)]
+pub struct ConnectionRequest {
```

Comment:
I'm a bit unclear about what this struct does--does this only get constructed when there is an access controller installed? Or only when a valid StreamID is passed?

Reply:
It always gets passed, similar in some way to an HttpRequest. Of course the response is limited to accept and reject though, which entirely informed by and sympathetic to the semantics of the protocol.